ambry-router/src/main/java/com/github/ambry/router/BatchOperationCallbackTracker.java

```diff
@@ -78,6 +84,9 @@
    * @param e the {@link Exception} that occurred (if any).
    */
   private void complete(Exception e) {
+    if(completionCallback != null) {
+      completionCallback.onCompletion(null, null);
+    }
```

Comment:
The callback will be invoked in line 91. we don't need another one.

Reply:
fixed.